src/modules/companies/validators/invoice-accounts.js

```diff
@@ -0,0 +1,106 @@
+const Joi = require('@hapi/joi');
+
+const CONTACT_TYPES = ['person', 'department'];
+const COMPANY_TYPES = ['individual', 'limitedCompany', 'partnership', 'limitedLiabilityPartnership'];
```

Comment:
I think the majority of this validation logic probably lives in the service models.

Where only that property is affected, the validation logic can be in the setters/getters.

Where you have dependencies between properties (e.g. in the address model), we could have a:

`isValidAddress()`

method to check the fully populated model.